.github/workflows/Dockerfile.yml

```diff
@@ -0,0 +1,41 @@
+name: Dockerfile
+on:
+  pull_request:
+    paths:
+      - "build/**"
+      - "Dockerfile"
+      - ".dockerignore"
+      - ".github/workflows/Dockerfile.yml"
+
+# Each PR and each commit on `main` use distinct concurrency groups. Note canceling workflows in
+# progress may result in having to rebuild layers as the Docker layer caching is only uploaded
+# once the image has been fully built. That said, new commits may invalidate these layers anyway.
+concurrency:
+  group: dockerfile-${{ github.ref_name }}-${{ github.ref == 'refs/heads/main' && github.run_number }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    name: Build
+    runs-on:
+      labels: ubuntu-20.04-16core
+    env:
+      # Reference the HEAD commit which triggered this workflow. By default PRs use a merge commit
+      SHA: ${{ github.event.pull_request.head.sha || github.sha }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ env.SHA }}  # Always checkout HEAD commit
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Build
+        uses: docker/build-push-action@v4
+        with:
+          # Use Path context to ensure that the `.git` directory is included in the image.
+          # https://github.com/docker/build-push-action#git-context
+          context: .
+          push: false
+          file: Dockerfile
+          # TODO: May want to switch to another cache type as GHA have a 10 GB limit
```

Comment:
what's the current image size out of interest? would there be any utility in printing that to CI so we can see if we're approaching the limit? (I can see the individual layher sizes in the CI output but not a total image size from what I can see)

Reply:
The image is currently 5.93GB (from my local build). I'll try to add that info to this workflow. I'll call out that we are allowed to go over the limit but GHA will eventually clean up older artifacts